shiny/_docstring.py

```diff
@@ -18,104 +41,130 @@ class DocStringWithExample(str):
     ...
 
 
+class ExampleWriterRegistry:
+    def __init__(self):
+        self._writer = None
+
+    def set_writer(self, func: F) -> F:
+        self._writer = func
+        return func
+
+    def write_example(self, app_files: list[str], **kwargs: dict[str, Any]) -> str:
+        if self._writer is None:
+            return self.default_writer(app_files)
+        return self._writer(app_files, **kwargs)
+
+    def default_writer(self, app_files: list[str]) -> str:
+        app_file = app_files[0]
+        with open(app_file) as f:
+            code = f.read()
+
+        return f"```.python\n{code.strip()}\n```\n"
+
+
+example_writer = ExampleWriterRegistry()
```

Comment:
```
class ExampleWriter:
    def write_example(...):
        ...

example_writer = ExampleWriter

if IS_QUARTODOC:
    class ShinyLiveExampleWriter(ExampleWriter):
        def write_example(...):
            ...
    
   example_writer = ShinyLiveExampleWriter()
```


```
example_writer.write_example(...)
```

Reply:
Thanks! Done in ebf47c6cd8cfeb4ee216768ecccf379dc016419e